### PR TITLE
[202012] [Mellanox] Read transceiver EEPROM via sdk sysfs

### PR DESCRIPTION
**Pull request by keboliu**
This PR depends on https://github.com/sonic-net/sonic-buildimage/pull/12531

#### Why I did it
1. ethtool is not able to read certain pages(eg. page 11h) of CMIS cables.
2. SDK provides a set of sysfs to expose the transceiver EEPROM, now we migrate from using ethtool to read these sysfs for transceiver EEPROM reading.

#### How I did it
1. replace ethtool with accessing the SDK sysfs for cable EEPROM reading.
2. Adjust the offset according to the SDK sysfs memory map.

#### How to verify it
run sonic-mgmt sfp-related regression test case.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py [hunk 1]
```diff
@@ -1355,47 +1336,23 @@ def get_tx_fault(self):
...
-                    tx_fault_list.append(tx_fault_data & 0x40 != 0)
-                    tx_fault_list.append(tx_fault_data & 0x80 != 0)
-
-        else:
-            offset = 256
-            dom_channel_monitor_raw = self._read_eeprom_specific_bytes((offset + SFP_CHANNL_STATUS_OFFSET), SFP_CHANNL_STATUS_WIDTH)
-            if dom_channel_monitor_raw is not None:
-                tx_fault_data = int(dom_channel_monitor_raw[0], 16)
-                tx_fault_list.append(tx_fault_data & 0x04 != 0)
+                tx_fault_list = [False]*8
```
Comment (prgeor) [inline platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:1351]: can you add a comment for why hardcoding here?

**Review by prgeor — APPROVED** [review]

File: platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py (hunk 1 above)
Comment (keboliu) [reply platform/mellanox/mlnx-platform-api/sonic_platform/sfp.py:1361]: added comments at the beginning of both functuions.

**Review by Junchao-Mellanox — APPROVED** [review]

**Comment by keboliu** [comment]
**Pending for new SDK upstream, please hold the merge first**

**Comment by liat-grozovik** [comment]
Note: converted to draft as it is pending new SDK PR to be upstream first.

**Pull request merged**